py2-deprecation.md

```diff
@@ -0,0 +1,134 @@
+- Feature Name: Python 2 Deprecation Schedule
+- Start Date: 2019-03-25
+- SEP Status: Draft
+- SEP PR: (leave this empty)
+- Salt Issue: (leave this empty)
+
+
+
+# Summary
+[summary]: #summary
+
+In light of Python 2.7 reaching its end of life (EOL) on Jan 1st 2020, Python 2
+will be deprecated from SaltStack no earlier then the Sodium release, that is
+either the Sodium release or a later release.
+
+
+# Motivation
+[motivation]: #motivation
+
+Python 2.7 will reach it’s official end of life (EOL) on [January 1st,
+2020](https://devguide.python.org/#branchstatus) and post that, it will no
+longer be supported by its maintainers. Consequently, SaltStack team will
+retire Python 2.7 support in SaltStack.
+
+The end of Python 2.7 support will enable SaltStack to focus its effort on
+better supporting Python 3.4+.
+
+[Python 2.x is legacy, Python 3.x is the present and future of the
+language](https://wiki.python.org/moin/Python2orPython3)
+
+
+# Design
+[design]: #detailed-design
+
+Python 2 will be deprecated from SaltStack no earlier than the Sodium release
+(tentative release date is Feb 2020), that is either the Sodium release or a
+later release. Keeping in mind SaltStack’s policy of announcing ‘end of
+support’ at least 2 major releases before the intended changes takes place,
+Sodium or a later release has been chosen as a suitable deadline for Python 2
+deprecation.
+
+**Migration:** After carefully evaluating the list of Python 3 versions
+supported by different operating systems and by also considering the EOL for
+all these operating systems [1]* (as denoted in below grid) and [EOL of Python
+branches](https://devguide.python.org/#branchstatus)[2]*, SaltStack team
+proposes to support Python 3.4 from Sodium or a later release. Versions of
+SaltStack released before the cutoff release will continue to support python
+2.7 until they reach end of life.
+
+[![Grid.png](https://i.postimg.cc/V62Wh9nY/Grid.png)](https://postimg.cc/xJymJz67)
+
+### Note 1
+
+SaltStack currently supports master and minions running on different
+versions of Python such as salt master running on Python 2, minion running on
+Python 3 and vice versa. This support will not change in this transition as a
+change in Python interpreter does not mandate a change in network protocol.
+
+### Note 2
+
+The version of OpenSSL that SaltStack will support will be determined
+by the Python version supported by SaltStack. Check the [Python SSL
+docs](https://docs.python.org/3/library/ssl.html) for the versions of OpenSSL
+supported by each Python version, and check the [pyOpenSSL release
+page](https://pypi.org/project/pyOpenSSL/#history) to identify the which
+versions of Python are supported by pyOpenSSL.
+
+### Note 3
+
+A future SEP will define a schedule for ongoing depreciation of Python versions as they also reach their end-of-life.
+
+### Note 4
+
+A future SEP will define the salt-ssh Python deprecation schedule.
```

Comment:
Does this mean that salt-ssh will be deprecated or that salt-ssh will continue to support python2 (which means all modules and states will still need to support it) until some point in the future?